organisaatio-ui/src/ValidationSchemas/YhteystietoLomakeSchema.ts

```diff
@@ -2,52 +2,145 @@ import Joi from 'joi';
 
 export const postinumeroSchema = Joi.string().regex(/^\d{5}$/);
 
-export default Joi.object({
-    'kieli_fi#1': Joi.object({
-        postiOsoite: Joi.string().required(),
-        postiOsoitePostiNro: postinumeroSchema.required(),
-        postiOsoiteToimipaikka: Joi.string().allow(''),
-        kayntiOsoite: Joi.any(),
-        kayntiOsoitePostiNro: Joi.any(),
-        kayntiOsoiteToimipaikka: Joi.any(),
-        puhelinnumero: Joi.string().allow(''),
-        email: Joi.string()
-            .email({ tlds: { allow: false } })
-            .required(),
-        www: Joi.string().allow(''),
-    }).when('osoitteetOnEri', {
-        is: true,
-        then: Joi.object({
-            kayntiOsoite: Joi.string(),
-            kayntiOsoitePostiNro: postinumeroSchema.required(),
+export default Joi.alternatives().try(
```

Comment:
hitosti toistoa tässä -> erittäin vaikea hahmottaa koodista onko kahdella tapauksella jotain eroa vai ei. tästä varmasti saisi selkeämmän esityksen palastelemalla ja destructuroinnilla.

Reply:
joo kiitos. Palastelin asiaa, niin olisiko nyt järkevämpi?